model/canvas.cc

```diff
@@ -640,7 +640,10 @@ namespace minsky
           newItem=group->copy();
         else
           {
-            newItem.reset(item->clone());
+			//  Use GodleyIcon's native clone() method. for tickets 1180/1183  
+            if (auto godley=dynamic_cast<GodleyIcon*>(item.get()))
+              newItem.reset(godley->clone());      			    
```

Comment:
Because clone() is a virtual method, these lines are strictly not needed (and indeed, add a small, probably impercetible, delay due to the unnecessary dynamic_cast.

Reply:
I tried to remove them, but then the wires would not be copied along with the Godley icons in the example 1Free.mky.

Reply:
Something is wrong, because this entire conditional is strictly equivalent to newItem.reset(item->clone());

Reply:
I removed group.reset() from godleyIcon::clone(), which allowed me to remove the GodleyVars struct, but as soon as I remove newItem.reset(godley->clone()), the wires are no longer copied along with the Godley icon in 1Free.mky. I don't what else to do. I worked 9.5 hours on feature 94 yesterday so that may have sapped all my mental faculties when I looked at this again last night. I will try to look at this again after I get feature 94 working today.